src/controller/java/AndroidICDClient.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ *   Copyright (c) 2024 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+/**
+ *    @file
+ *      Implementation of ICD Client API for Android Platform
+ *
+ */
+
+#include "AndroidICDClient.h"
+
+#include <app/icd/client/ICDClientInfo.h>
+
+chip::app::DefaultICDClientStorage sICDClientStorage;
+
+jobject getICDClientInfo(JNIEnv * env, const char * icdClientInfoSign, jint jFabricIndex)
```

Comment:
although it is good to have getICDClientInfo to dump all lit client info for disgnostic purpose as we previous discussed, it may be better to introduce another getICDClientInfo with fabric index and nodeId for "isPeerICDClient usage" without stream match.

Reply:
@yunhanw-google 
The API providing ICDClientInfo already exists, so I don't think it's necessary to create a new one. (Previously, it was implemented in the ChipDeviceController class, and it was created as a separate class to use in InteractionClient.)

Reply:
ok, thanks